### ISSUE TITLE
Unable to reproduce the results reported in the paper

### ISSUE THREAD

**Issue by quynhkhanh96** [MrChenFeng/Adaptive-Soft-Contrastive-Learning_ICPR2022#3]
Hello, thanks for providing the implementation of ASCL.
I'm having trouble to reproduce the results reported in the paper. In particular, I ran experiments with small-scale datasets (CIFAR10, CIFAR100, TinyImageNet) with the following configuration:

### **Pretraining**
- Augmentation: Weak augmentation
- Memory bank size = 4096 (`K = 4096`)
- Updating momentum for memory encoder = 0.99 (`m = 0.99`)
- $\tau = 0.1$ (`t1 = 0.1`)
- SGD: momentum = 0.9, weight decay = 1e-4, lr = 0.06, cosine annealing scheduler
- Epochs = 200
### **Evaluation**
- Freeze the parameters of the encoder and train the linear classifier without the projection head.
- Epochs = 100
- SGD: momentum = 0.9, no weight decay, lr = 10., reduced to 1. at epoch 60 and to 0.1 at epoch 80.

```shell
python main_simple.py  --epochs 200 --gpuid 0 --dataset cifar100 --data_path data/CIFAR100 --lr 0.06 --wd 1e-4 --K 4096 --model moco --type normal --t1 0.1 --aug weak_augment
```

### **Pretraining**
- Augmentation: Weak augmentation
- Memory bank size = 4096 (`K = 4096`)
- Updating momentum for memory encoder = 0.99 (`m = 0.99`)
- `nn_num = 2` (number of top closest samples) 
- $\tau = 0.1, \tau^{'} = 0.05$ (`t1 = 0.1`, `t2 = 0.05`)
- SGD: momentum = 0.9, weight decay = 1e-4, lr = 0.06, cosine annealing scheduler
- Epochs = 200
### **Evaluation**
- Freeze the parameters of the encoder and train the linear classifier without the projection head.
- Epochs = 100
- SGD: momentum = 0.9, no weight decay, lr = 10., reduced to 1. at epoch 60 and to 0.1 at epoch 80.

```shell
python main_simple.py  --epochs 200 --gpuid 0 --dataset cifar100 --data_path data/CIFAR100 --lr 0.06 --wd 1e-4 --K 4096 --nn_num=2  --model moco --type ascl --t1 0.1 --t2 0.05 --aug weak_augment
```

Here are the results I got: 
| Dataset | Method | Reported    | Replicated   |
| :---:   | :---: | :---: | :---: |
| CIFAR100 | MoCo | 64.02 (table III) |  60.72  |
| CIFAR100 | ASCL (nn_num=2) | 65.35 (table III) |  62.53  |
| CIFAR10 | MoCo | 88.55 (table III) | 85.83 |
| CIFAR10 |  ASCL (nn_num=2)|  90.09 (table III) | 87.46 |
| TinyImageNet | MoCo | 43.36 (table I) | 45.91 |
| TinyImageNet |  ASCL (nn_num=2)| 48.36 (table I) | 45.90 |

It can be observed that even though ASCL mostly yields better downstream classification accuracy than baseline MoCo, the results are not as high as reported in the paper. Please help me with that. Thank you in advance!

**Comment by MrChenFeng** [comment]
Hi,

I checked my cached logs, it looks okay.
https://api.wandb.ai/links/mrchenfeng/fc2utswv
I'm afraid I can't recall too many about this repo currently since it's too old. I would suggest to try with **exactly the example runs** in readme.

Chen

**Comment by quynhkhanh96** [comment]
Hi, thanks for your quick reply. The example in the readme looks strange to me as it would set `K` to 5, instead of 4096 as stated in the paper (the memory bank size).

**Comment by MrChenFeng** [comment]
Please use the updated `main_simple.py`, K is the number of neighbors. There exist unintentional typo when I try to make the `main_simple.py` for easier reproduction.
